src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -82,6 +85,49 @@ struct SimplexResult
     mpsWriterFactory mps_writer_factory;
 };
 
+static void fillModelerComponents(std::vector<std::unique_ptr<ComponentFiller>>& componentFillers,
+                                  std::vector<LinearProblemFiller*>& fillersCollection,
+                                  const Antares::Study::SystemModel::System* modelerSystem)
+{
+    if (!modelerSystem)
+    {
+        return;
+    }
+
+    for (const auto& [_, component]: modelerSystem->Components())
+    {
+        componentFillers.push_back(std::make_unique<ComponentFiller>(component));
+    }
+    for (auto& component_filler: componentFillers)
+    {
+        fillersCollection.push_back(component_filler.get());
+    }
+}
+
+static void writeModelerSolutions(const operations_research::MPSolver* solver,
+                                  Optimization::PROBLEME_SIMPLEXE_NOMME& Probleme,
+                                  const int optimizationNumber,
+                                  const OptPeriodStringGenerator& optPeriodStringGenerator,
+                                  IResultWriter& writer)
+{
+    std::stringstream contentStream;
+    std::vector<MPVariable*> variables = solver->variables();
```

Comment:
You don't need to copy the std::vector, because this method returns a const reference
```cpp
  const std::vector<MPVariable*>& variables() const { return variables_; }
```